src/mlpack/tests/main_tests/kmeans_test.cpp

```diff
@@ -0,0 +1,408 @@
+ */
+BOOST_AUTO_TEST_CASE(KmClusteringSizeCheckLabelOnly)
+{
+  int C = 2;
+
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+      BOOST_FAIL("Unable to load train dataset vc2.csv!");
+  size_t col = inputData.n_cols;
+  size_t row = inputData.n_rows;
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("clusters", std::move(C));
+  SetInputParam("labels_only", true);
+  SetInputParam("allow_empty_clusters", false);
+
+
+  mlpackMain();
+
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_rows, 1);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_cols, col);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_rows, row);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_cols, C);
+}
+
+
+/**
+ * Checking that predictions are not same when --allow_empty_clusters or kill_empty_clusters are specified
+ */
+BOOST_AUTO_TEST_CASE(KmClusteringEmptyClustersCheck)
+{
+  int C = 400;
+  int iterations = 100;
+
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+      BOOST_FAIL("Unable to load train dataset vc2.csv!");
+
+
+  SetInputParam("input", inputData);
+  SetInputParam("clusters", C);
+  SetInputParam("labels_only", true);
+  SetInputParam("max_iterations", iterations);
+  SetInputParam("allow_empty_clusters", false);
+
+  mlpackMain();
+
+  arma::mat normalOutput;
+  normalOutput = std::move(CLI::GetParam<arma::mat>("output"));
+
+  ResetKmSettings();
+
+  SetInputParam("input", inputData);
+  SetInputParam("clusters", C);
+  SetInputParam("labels_only", true);
+  SetInputParam("allow_empty_clusters", true);
+  SetInputParam("max_iterations", iterations);
+
+  mlpackMain();
+
+  arma::mat allowEmptyOutput;
+  allowEmptyOutput = std::move(CLI::GetParam<arma::mat>("output"));
+
+  ResetKmSettings();
+
+  SetInputParam("input", inputData);
+  SetInputParam("clusters", std::move(C));
+  SetInputParam("labels_only", true);
+  SetInputParam("kill_empty_clusters", true);
+  SetInputParam("max_iterations", std::move(iterations));
+
+  mlpackMain();
+
+  arma::mat killEmptyOutput;
+  killEmptyOutput = std::move(CLI::GetParam<arma::mat>("output"));
+
+  ResetKmSettings();
+
+  BOOST_REQUIRE_GT(arma::accu(killEmptyOutput != allowEmptyOutput), 1);
+  BOOST_REQUIRE_GT(arma::accu(normalOutput != allowEmptyOutput), 1);
+}
+
+/**
+ * Ensuring that absence of Input is checked.
+ */
+BOOST_AUTO_TEST_CASE(KmNoInputData)
+{
+  int C = 2;
+
+  SetInputParam("clusters", std::move(C));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Checking that that size and dimensionality of Final Input File is correct
+ * when flag --in_place is specified
+ */
+BOOST_AUTO_TEST_CASE(KmClusteringResultSizeCheck)
+{
+  int C = 2;
+
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+      BOOST_FAIL("Unable to load train dataset vc2.csv!");
```

Comment:
This should only be indented once (this issue occurs in several other places too).